ocs_ci/ocs/resources/objectbucket.py

```diff
@@ -0,0 +1,388 @@
+
+
+class MCGS3Bucket(ObjectBucket):
+    """
+    Implementation of an MCG bucket using the S3 API
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.mcg.s3_resource.create_bucket(Bucket=self.name)
+
+    def internal_delete(self):
+        """
+        Deletes the bucket using the S3 API
+        """
+        self.mcg.s3_resource.Bucket(self.name).object_versions.delete()
+        self.mcg.s3_resource.Bucket(self.name).delete()
+
+    @property
+    def internal_status(self):
+        """
+        Returns the OBC mode as shown in the NB UI and retrieved via RPC
+
+        Returns:
+            str: The bucket's mode
+
+        """
+        return self.mcg.get_bucket_info(self.name).get('mode')
+
+    def internal_verify_health(self):
+        """
+        Verifies that the bucket is healthy by checking its mode
+
+        Returns:
+            bool: True if the bucket is healthy, False otherwise
+
+        """
+        return self.status == constants.HEALTHY_OB
+
+    def internal_verify_deletion(self):
+        return self.name not in self.mcg.s3_get_all_bucket_names()
+
+
+class OCBucket(ObjectBucket):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def internal_delete(self, verify=True):
+        """
+        Deletes the bucket using the OC CLI
+        """
+        OCP(kind='obc', namespace=self.namespace).delete(resource_name=self.name)
+
+    @property
+    def internal_status(self):
+        """
+        Returns the OBC's phase
+
+        Returns:
+            str: OBC phase
+
+        """
+        return OCP(
+            kind='obc', namespace=self.namespace, resource_name=self.name
+        ).get()['status']['phase']
+
+    def internal_verify_health(self):
+        """
+        Verifies that the bucket is healthy by checking its phase
+
+        Returns:
+            bool: True if the bucket is healthy, False otherwise
+
+        """
+        return self.status == constants.HEALTHY_OBC
+
+    def internal_verify_deletion(self):
+        return self.name not in oc_get_all_obc_names()
+
+
+class MCGOCBucket(OCBucket):
+    """
+    Implementation of an MCG bucket using the OC CLI
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        obc_data = templating.load_yaml(constants.MCG_OBC_YAML)
+        if self.name is None:
+            self.name = create_unique_resource_name('oc', 'obc')
+        obc_data['metadata']['name'] = self.name
+        obc_data['spec']['bucketName'] = self.name
+        obc_data['spec']['storageClassName'] = self.namespace + '.noobaa.io'
+        obc_data['metadata']['namespace'] = self.namespace
+        if 'bucketclass' in kwargs:
+            obc_data.setdefault('spec', {}).setdefault('additionalConfig', {}).setdefault(
+                'bucketclass', kwargs['bucketclass']
+            )
+        create_resource(**obc_data)
```

Comment:
sorry for commenting on this topic once again. I think that class constructors should not do actual resource creation. IMO, we should keep the consistency of the class the instantiation and the resource creation separated.

Reply:
For the record -
We had a call regarding the matter of OBC instantiation creating an OBC instead of possibly retrieving an existing one.

Further discussions are needed in order to better understand the current situation and to think of the best way to proceed further with the matter.

I'm resolving this and documenting the matter under #2573